File: Stack/Stack.java
package dataStructures.Stack;

import java.util.ArrayList;

public class Stack {

    public ArrayList<Integer> arr = new ArrayList<Integer>();

    public void insert(int item) {
        this.arr.add(item);
    }

    public void delete() {
        if (!this.isEmpty()) {
            this.arr.remove(this.arr.size() - 1);
        }
    }

    public int peek() {
        return this.arr.get(this.arr.size() - 1);
    }

    public boolean isEmpty() {
        return this.arr.isEmpty();
    }

    public void print() {
        System.out.println("\n------------- Printing ---------");
        for (int i = this.arr.size() - 1; i >= 0; i--) {
            System.out.println(this.arr.get(i) + "\n");
        }
    }
}

File: Stack/stack.go
package main

import "fmt"

type Stack []int

func (s *Stack) Insert(item int) {
	*s = append(*s, item)
}

func (s *Stack) Delete() {
	if !s.IsEmpty() {
		*s = (*s)[:len(*s)-1]
	}
}

func (s *Stack) IsEmpty() bool {
	return len(*s) == 0
}

func (s *Stack) Peek() int {
	return (*s)[len(*s)-1]
}

func (s *Stack) Print() {
	fmt.Println("\n ---------- Printing -------- ")
	for i := len(*s) - 1; i >= 0; i-- {
		fmt.Println((*s)[i])
	}
}

func main() {
	var stack Stack

	stack.Insert(25)
	stack.Insert(34)
	stack.Insert(12)
	stack.Insert(3)
	stack.Insert(87)
	stack.Insert(43)
	stack.Insert(29)
	stack.Print()
	stack.Delete()
	stack.Print()
	stack.Peek()
}
